File: 04 Technical/02 Server/01 Logics/01 Python/01 General/07 Armstrong Number.py

# ArmStrong Number

given_number = "153"  # input()
print(given_number)

print("-------------------")

power = len(given_number)

result = 0
for each_char in given_number:
    each_num = int(each_char)
    add_value = each_num ** power
    result += add_value

is_armstrong = int(given_number) == result

print(is_armstrong)  # True

print("======================")

File: 04 Technical/01 Client/01 Logics/01 Javascript/07 Armstrong Number.js
// Armstrong Number

let given_number = '153';
console.log(given_number);

console.log('-------------------');

let power = given_number.length;

let is_armstrong = false;

result = 0;
for (each_number of given_number) {
	add_value = parseInt(each_number) ** power;
	result += add_value;
}

if (result == parseInt(given_number)) {
	is_armstrong = true;
}

console.log(is_armstrong);

console.log('=======================');
